### PR TITLE
refactor: header menu

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

File: src/angular/header-lean/header-menu-item.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+} from '@angular/core';
+import { mixinDisabled, TypeRef } from '@sbb-esta/angular/core';
+
+// Boilerplate for applying mixins to HeaderMenuItemDirective.
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu-item.ts:16]:
```suggestion
// Boilerplate for applying mixins to SbbHeaderMenuItem.
```

File: src/angular/header-lean/header-menu-item.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        if (element.childNodes[i].nodeType === textNodeType) {
+          output += element.childNodes[i].textContent;
+        }
+      }
+    }
+
+    return output.trim();
+  }
+
+  // tslint:disable: member-ordering
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu-item.ts:115]:
```suggestion
```

File: src/angular/header-lean/header-menu-item.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  static ngAcceptInputType_disabled: BooleanInput;
+  // tslint:enable: member-ordering
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu-item.ts:117]:
```suggestion
```

File: src/angular/header-lean/header-menu-trigger.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+  provide: SBB_HEADER_MENU_SCROLL_STRATEGY,
+  deps: [Overlay],
+  useFactory: SBB_HEADER_MENU_SCROLL_STRATEGY_FACTORY,
+};
+
+/** Options for binding a passive event listener. */
+const passiveEventListenerOptions = normalizePassiveListenerOptions({ passive: true });
+
+@Component({
+  // tslint:disable-next-line: component-selector
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu-trigger.ts:55]:
```suggestion
```

File: src/angular/header-lean/header-menu-trigger.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+  selector: 'button[sbbHeaderMenu]',
+  templateUrl: './header-menu-trigger.html',
+  styleUrls: ['./header-menu-trigger.css'],
+  // changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu-trigger.ts:59]: Maybe a comment why commented?

File: src/angular/header-lean/header-menu.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<button type="button" class="sbb-header-menu-back" (click)="open = false">
+  <sbb-icon svgIcon="kom:chevron-small-left-small"></sbb-icon>
+  <span i18n="Go back to the app chooser navigation@@sbbHeaderMenuBack">Back</span>
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header-menu.html:3]: use localize insteaf of i18n

File: src/angular/header-lean/header.ts [hunk 7]
```diff
@@ -311,6 +334,18 @@ export class SbbHeaderLean implements OnChanges, AfterViewInit, OnDestroy {
     }
   }
 
+  /*
```
Comment (jeripeierSBB) [inline src/angular/header-lean/header.ts:337]: still needed?

File: src/angular/menu/menu.scss [hunk 8]
```diff
@@ -32,7 +32,7 @@ sbb-menu {
     pointer-events: none;
```
Comment (jeripeierSBB) [inline src/angular/menu/menu.scss:32]: is this not too risky to change in this PR?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Generally ok, but a few things to optimize

File: src/angular/menu/menu.scss (hunk 8 above)
Comment (jeripeierSBB) [reply src/angular/menu/menu.scss:32]: (I mean all :where statements)

Comment (kyubisation) [reply src/angular/menu/menu.scss:32]: That was unintentional

**Pull request merged**